matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/DefaultFileService.kt

```diff
@@ -145,15 +151,21 @@ internal class DefaultFileService @Inject constructor(
                     Timber.v("## FileService: decrypt file")
                     // Ensure the parent folder exists
                     cachedFiles.decryptedFile.parentFile?.mkdirs()
+                    // Write to a tmp file first, so if we abort before done, we don't have a broken cached file
+                    val tmpFile = File(cachedFiles.decryptedFile.parentFile, "${cachedFiles.decryptedFile.name}.tmp")
```

Comment:
Create a helper to avoid repeated the same code twice?
For instance
```kotlin
internal class AtomicFileCreator(private val file: File) {
    val tmpFile = File(file.parentFile, "${file.name}.tmp")

    init {
        if (tmpFile.exists()) {
            Timber.v("## AtomicFileCreator: discard aborted tmp file ${tmpFile.path}")
        }
    }

    fun commit() {
        tmpFile.renameTo(file)
    }
}
```